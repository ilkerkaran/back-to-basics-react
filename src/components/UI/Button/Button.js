import React from 'react';

import './Button.css';
import PropTypes from 'prop-types';

const button = props => {
  return (
    <button
      className={['Button', props.buttonType].join(' ')}
      onClick={props.onClick}
    >
      {props.children}
    </button>
  );
};

export default button;

button.propTypes = {
  buttonType: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};
