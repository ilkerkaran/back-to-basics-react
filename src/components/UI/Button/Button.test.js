import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';


describe('<Button /> snapshot ', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Button buttonType ="Succes" inputType="button" onClick={() => {}}>Clcik Me</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
