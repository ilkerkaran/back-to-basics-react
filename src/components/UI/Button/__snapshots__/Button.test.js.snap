// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> snapshot  renders correctly 1`] = `
<button
  className="Button Succes"
  onClick={[Function]}
  type="button"
>
  Clcik Me
</button>
`;
